relayer/relayer.go

```diff
@@ -217,6 +226,21 @@ func (r *Relayer) processMissedBlocks(
 	return err
 }
 
+// Sets the relayer health status to false while attempting to reconnect.
+func (r *Relayer) ReconnectToSubscriber() error {
+	r.healthStatus.Store(false)
+
+	// Attempt to reconnect the subscription
+	err := r.Subscriber.Subscribe(maxResubscribeAttempts)
```

Comment:
I think we should still limit the number of resubscribe attempts, even if we've successfully subscribed to the node in the past. Resubscription is no more likely to succeed than initial subscription, so we should treat them the same.

Reply:
That said, if we drop a connection to a node, we should refresh any values fetched from that node on startup, such as the ChainConfig. I think that would involve a significant refactor though, so let's save it for a future change.

Reply:
I do think the initial connection and reconnection are distinctly different. If you fail to ever connect, you may have something misconfigured so you will never be able to connect, and exiting the application quickly is probably a good idea. 

If you're trying to reconnect, everything was set up correctly the first time, and something has probably happened to the node. I think we should keep trying to reconnect in case it comes back up, and also not exit the program, because all the other subscribers are also working. We set the unhealthy flag for the reconnecting subscriber, so someone can kill the node if they see it is unhealthy and think that a restart is the correct option.

Reply:
Yeah, I agree we should re-fetch values when trying to reconnect. It shouldn't be too bad of a refactor once the open PRs here have landed.

Reply:
How about this:
- Let's attempt to resubscribe the same number of times as initial subscription here, and exit the program if we're unable to do so.
- When we implement re-fetching values on successful resubscription, we can relax the resubscription logic to continue to attempt to resubscribe in perpetuity.

My main concern with the proposed change as-is is that when we eventually resubscribe, values might be stale. Quitting early and forcing the program to restart bypasses this.

Reply:
Which values in the relayer are stateful? Is it just the Quorum numbers? As-is right now we also won't process blocks missed between disconnection and reconnection, but with the change I asked for in your PR that will be an easy change.

I'm okay to defer both those changes to a future PR if we lower the resubscribe attempts as you suggested here and land this sooner.

Reply:
Yup, for now that's the only state we would need to refresh. But that set could easily grow in the future. I'm envisioning a unified `FetchSubnetValues` method that could be called at startup, on resubscribing, and periodically in the background to keep everything up to date. Then we can easily add in fetched state values in a single place.